source/2_platform/rex_opengl/src/resources/vertex.cpp

```diff
@@ -1,248 +1,228 @@
-        };
-
-        return buffer_layouts;
-    }
-}
\ No newline at end of file
+  //-----------------------------------------------------------------------
+  // VERTEX POS
+  //
+  // ATTRIBUTES:
+  //  - POSITION
+  //-----------------------------------------------------------------------
+  BufferLayout VertexPos::layout()
+  {
+    BufferLayoutBuilder builder;
+    builder.add_buffer_element(DataType::Value::Vec3);
+    return builder.build();
+  }
+
+  //-----------------------------------------------------------------------
+  VertexPos::VertexPos()
+      : m_position(0.0f)
+  {
+  }
+
+  //-----------------------------------------------------------------------
+  VertexPos::VertexPos(const glm::vec3& position)
+      : m_position(position)
+  {
+  }
+
+  //-----------------------------------------------------------------------
+  const glm::vec3& VertexPos::position() const
+  {
+    return m_position;
+  }
+
+  //-----------------------------------------------------------------------
+  bool operator==(const VertexPos& v1, const VertexPos& v2)
+  {
+    return v1.position().x == v2.position().x && v1.position().y == v2.position().y && v1.position().z == v2.position().z;
+  }
+  //-----------------------------------------------------------------------
+  bool operator!=(const VertexPos& v1, const VertexPos& v2)
+  {
+    return !(v1 == v2);
+  }
+
+  //-----------------------------------------------------------------------
+  // VERTEX POS COL
+  //
+  // ATTRIBUTES:
+  //  - POSITION
+  //  - COLOR
+  //-----------------------------------------------------------------------
+  BufferLayout VertexPosCol::layout()
+  {
+    BufferLayoutBuilder builder;
+    builder.add_buffer_element(DataType::Value::Vec3);
+    builder.add_buffer_element(DataType::Value::Vec3);
+    return builder.build();
+  }
+
+  //-----------------------------------------------------------------------
+  VertexPosCol::VertexPosCol()
+      : m_position(0.0f)
+      , m_color(0.0f)
+  {
+  }
+
+  //-----------------------------------------------------------------------
+  VertexPosCol::VertexPosCol(const glm::vec3& position, const glm::vec3& color)
+      : m_position(position)
+      , m_color(color)
+  {
+  }
+
+  //-----------------------------------------------------------------------
+  const glm::vec3& VertexPosCol::position() const
+  {
+    return m_position;
+  }
+  //-----------------------------------------------------------------------
+  const glm::vec3& VertexPosCol::color() const
+  {
+    return m_color;
+  }
+
+  //-----------------------------------------------------------------------
+  bool operator==(const VertexPosCol& v1, const VertexPosCol& v2)
+  {
+    return v1.position().x == v2.position().x && v1.position().y == v2.position().y && v1.position().z == v2.position().z && v1.color().x == v2.color().x && v1.color().y == v2.color().y && v1.color().z == v2.color().z;
```

Comment:
clang-format put it like this, if you want to keep the old format, you'll have to disable clang-format for his function